### PR TITLE
#340: Create htmlcomment macro bridge

### PR DESCRIPTION
**Pull request by Josue-T**
Implement `htmlcomment` macro.

cf: #340

### CODE REVIEW

File: xwiki-pro-macros-confluence-bridges/xwiki-pro-macros-confluence-bridges-ui/src/main/resources/Confluence/Macros/HtmlComment.xml [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+
+<xwikidoc version="1.5" reference="Confluence.Macros.HtmlComment" locale="">
```
Comment (trrenty) [inline xwiki-pro-macros-confluence-bridges/xwiki-pro-macros-confluence-bridges-ui/src/main/resources/Confluence/Macros/HtmlComment.xml:23]: Did you export the page using a 14.10 xwiki instance?

File: xwiki-pro-macros-confluence-bridges/xwiki-pro-macros-confluence-bridges-ui/src/main/resources/Confluence/Macros/HtmlComment.xml [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+  <language/>
+  <defaultLanguage/>
+  <translation>0</translation>
+  <creator>xwiki:XWiki.Admin</creator>
+  <parent>WebHome</parent>
+  <author>xwiki:XWiki.Admin</author>
+  <originalMetadataAuthor>XWiki.Admin</originalMetadataAuthor>
+  <contentAuthor>xwiki:XWiki.Admin</contentAuthor>
+  <version>1.1</version>
+  <title>HtmlComment</title>
```
Comment (trrenty) [inline xwiki-pro-macros-confluence-bridges/xwiki-pro-macros-confluence-bridges-ui/src/main/resources/Confluence/Macros/HtmlComment.xml:35]: For the sake of consistency, you should prefix the name of the page with `Confluence`

File: xwiki-pro-macros-confluence-bridges/xwiki-pro-macros-confluence-bridges-ui/src/main/resources/Confluence/Macros/HtmlComment.xml [hunk 3]
```diff
@@ -0,0 +1,405 @@
...
+  <comment/>
+  <minorEdit>false</minorEdit>
+  <syntaxId>xwiki/2.1</syntaxId>
+  <hidden>true</hidden>
+  <content>The ##htmlcomment## macro is a bridge between Confluence and XWiki. It give to possibility to add comment in HTML code.
```
Comment (trrenty) [inline xwiki-pro-macros-confluence-bridges/xwiki-pro-macros-confluence-bridges-ui/src/main/resources/Confluence/Macros/HtmlComment.xml:40]:
```suggestion
  <content>The ##htmlcomment## macro is a bridge between Confluence and XWiki. It gives the possibility to add a comment in HTML code.
```

File: xwiki-pro-macros-confluence-bridges/xwiki-pro-macros-confluence-bridges-ui/src/main/resources/Confluence/Macros/HtmlComment.xml [hunk 4]
```diff
@@ -0,0 +1,405 @@
...
+    <property>
+      <async_context/>
+    </property>
+    <property>
+      <async_enabled>0</async_enabled>
+    </property>
+    <property>
+      <code>{{velocity}}
+{{html clean="false"}}
+#if ("$!wikimacro.parameters.get('hide-in-source-code')" == '' || ! "$!wikimacro.parameters.get('hide-in-source-code')")
```
Comment (trrenty) [inline xwiki-pro-macros-confluence-bridges/xwiki-pro-macros-confluence-bridges-ui/src/main/resources/Confluence/Macros/HtmlComment.xml:277]: What is the second condition checking?

File: xwiki-pro-macros-confluence-bridges/xwiki-pro-macros-confluence-bridges-ui/src/main/resources/Confluence/Macros/HtmlComment.xml [hunk 5]
```diff
@@ -0,0 +1,405 @@
...
+      <defaultCategories/>
+    </property>
+    <property>
+      <description>Macro bridge for confluence HTML Comment macro</description>
+    </property>
+    <property>
+      <id>htmlcomment</id>
+    </property>
+    <property>
+      <name>HTML Comment</name>
```
Comment (trrenty) [inline xwiki-pro-macros-confluence-bridges/xwiki-pro-macros-confluence-bridges-ui/src/main/resources/Confluence/Macros/HtmlComment.xml:303]:
```suggestion
      <name>Confluence Bridge for HTML Comment</name>
```

File: xwiki-pro-macros-confluence-bridges/xwiki-pro-macros-confluence-bridges-ui/src/main/resources/Confluence/Macros/HtmlComment.xml [hunk 6]
```diff
@@ -0,0 +1,405 @@
...
+      <contentJavaType>Wiki</contentJavaType>
+    </property>
+    <property>
+      <contentType>Optional</contentType>
+    </property>
+    <property>
+      <defaultCategories/>
+    </property>
+    <property>
+      <description>Macro bridge for confluence HTML Comment macro</description>
```
Comment (trrenty) [inline xwiki-pro-macros-confluence-bridges/xwiki-pro-macros-confluence-bridges-ui/src/main/resources/Confluence/Macros/HtmlComment.xml:297]:
```suggestion
      <description>Confluence Bridge for HTML Comment macro.</description>
```

**Review by trrenty — CHANGES_REQUESTED** [review]

File: xwiki-pro-macros-confluence-bridges/xwiki-pro-macros-confluence-bridges-ui/src/main/resources/Confluence/Macros/HtmlComment.xml (hunk 1 above)
Comment (Josue-T) [reply xwiki-pro-macros-confluence-bridges/xwiki-pro-macros-confluence-bridges-ui/src/main/resources/Confluence/Macros/HtmlComment.xml:23]: Now I developed with a 16.04 instance. Should I re-export from a 14.10 instance ?

File: xwiki-pro-macros-confluence-bridges/xwiki-pro-macros-confluence-bridges-ui/src/main/resources/Confluence/Macros/HtmlComment.xml (hunk 4 above)
Comment (Josue-T) [reply xwiki-pro-macros-confluence-bridges/xwiki-pro-macros-confluence-bridges-ui/src/main/resources/Confluence/Macros/HtmlComment.xml:277]: it was to check that the boolean value of `hide-in-source-code` is set to false.

File: xwiki-pro-macros-confluence-bridges/xwiki-pro-macros-confluence-bridges-ui/src/main/resources/Confluence/Macros/HtmlComment.xml (hunk 1 above)
Comment (trrenty) [reply xwiki-pro-macros-confluence-bridges/xwiki-pro-macros-confluence-bridges-ui/src/main/resources/Confluence/Macros/HtmlComment.xml]: Yes. The current parent of the pro-macros is 13.10. However, we need to upgrade it to 14.10 asap. It should be ok for you export this macro from a 14.10 instance

File: xwiki-pro-macros-confluence-bridges/xwiki-pro-macros-confluence-bridges-ui/src/main/resources/Confluence/Macros/HtmlComment.xml (hunk 4 above)
Comment (trrenty) [reply xwiki-pro-macros-confluence-bridges/xwiki-pro-macros-confluence-bridges-ui/src/main/resources/Confluence/Macros/HtmlComment.xml]: In this case, i think you need to unwrap the condition from the quotes. 

This code snipped displays `Hello2`. The strings "false"/"true" do not seem to be converted into booleans. The ! operator seems to only work on booleans.

```
{{velocity}}
#set ($smth = false)
#if (! "$!smth")
 Hello1
#end

#if (!$smth)
 Hello2
#end
{{/velocity}}
```

Actually. I think you can simply check `"$!wikimacro.parameters.get('hide-in-source-code')" == 'false'`

File: xwiki-pro-macros-confluence-bridges/xwiki-pro-macros-confluence-bridges-ui/src/main/resources/Confluence/Macros/ConfluenceHtmlComment.xml [hunk 7]
```diff
@@ -58,7 +58,7 @@ Result HTML:
 &lt;!-- it s a comment --&gt;
 {{/code}}</content>
   <object>
-    <name>Confluence.Macros.HtmlComment</name>
+    <name>Confluence.Macros.ConfluenceHtmlCommentt</name>
```
Comment (trrenty) [inline xwiki-pro-macros-confluence-bridges/xwiki-pro-macros-confluence-bridges-ui/src/main/resources/Confluence/Macros/ConfluenceHtmlComment.xml:61]:
```suggestion
    <name>Confluence.Macros.ConfluenceHtmlComment</name>
```

File: xwiki-pro-macros-confluence-bridges/xwiki-pro-macros-confluence-bridges-ui/src/main/resources/Confluence/Macros/ConfluenceHtmlComment.xml [hunk 8]
```diff
@@ -313,7 +313,7 @@ Result HTML:
     </property>
   </object>
   <object>
-    <name>Confluence.Macros.HtmlComment</name>
+    <name>Confluence.Macros.ConfluenceHtmlCommentt</name>
```
Comment (trrenty) [inline xwiki-pro-macros-confluence-bridges/xwiki-pro-macros-confluence-bridges-ui/src/main/resources/Confluence/Macros/ConfluenceHtmlComment.xml:316]:
```suggestion
    <name>Confluence.Macros.ConfluenceHtmlComment</name>
```

File: xwiki-pro-macros-confluence-bridges/xwiki-pro-macros-confluence-bridges-ui/src/main/resources/Confluence/Macros/HtmlComment.xml (hunk 4 above)
Comment (Josue-T) [reply xwiki-pro-macros-confluence-bridges/xwiki-pro-macros-confluence-bridges-ui/src/main/resources/Confluence/Macros/HtmlComment.xml]: Fixed thanks for the explanation.

File: xwiki-pro-macros-confluence-bridges/xwiki-pro-macros-confluence-bridges-ui/src/main/resources/Confluence/Macros/HtmlComment.xml (hunk 1 above)
Comment (Josue-T) [reply xwiki-pro-macros-confluence-bridges/xwiki-pro-macros-confluence-bridges-ui/src/main/resources/Confluence/Macros/HtmlComment.xml]: Did a import/export from a 14.10 instance. I still have the `xwikidoc version="1.5"` but some class fields was updated.

**Pull request merged**